WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -2312,6 +2359,43 @@ - (BOOL)mediaPickerController:(WPMediaPickerViewController *)picker shouldSelect
     return NO;
 }
 
+#pragma mark - UIDocumentPickerDelegate
+
+- (void)documentPicker:(UIDocumentPickerViewController *)controller didPickDocumentsAtURLs:(NSArray<NSURL *> *)urls
+{
+    if (urls.count == 0) { return; }
+    
+    [self.editorView.contentField focus];
+    [self prepareMediaProgressForNumberOfAssets:urls.count];
+    
+    for (NSURL *url in urls) {
+        [self addExternalMediaWithURL:url];
+    }
+}
+
+- (void)addExternalMediaWithURL:(NSURL *)url
+{
+    MediaService *mediaService = [[MediaService alloc] initWithManagedObjectContext:[[ContextManager sharedInstance] mainContext]];
+    
+    NSString *mediaUniqueID = [url lastPathComponent];
+    BOOL isImage = [WPImageViewController isUrlSupported:url];
+    __weak __typeof__(self) weakSelf = self;
+    [mediaService createMediaWithURL:url
+                     forPostObjectID:self.post.objectID
+                   thumbnailCallback:^(NSURL *thumbnailURL) {
+                       __typeof__(self) strongSelf = weakSelf;
+                       if (strongSelf) {
```

Comment:
As mentioned above, I don't think we need to worry about strong self here. (someone else please tell me if I'm wrong!)